lib/openapi.js

```diff
@@ -0,0 +1,194 @@
+    } else {
+      openapiObject.components = { schemas: {} }
+    }
+    if (tags) {
+      openapiObject.tags = tags
+    }
+    if (externalDocs) {
+      openapiObject.externalDocs = externalDocs
+    }
+
+    for (const [key, value] of extensions) {
+      openapiObject[key] = value
+    }
+
+    ref = Ref()
+    openapiObject.components.schemas = {
+      ...openapiObject.components.schemas,
+      ...(ref.definitions().definitions)
+    }
+
+    // Swagger doesn't accept $id on /definitions schemas.
+    // The $ids are needed by Ref() to check the URI so we need
+    // to remove them at the end of the process
+    Object.values(openapiObject.components.schemas)
+      .forEach(_ => { delete _.$id })
+
+    openapiObject.paths = {}
+
+    for (const route of routes) {
+      const schema = transform
+        ? transform(route.schema)
+        : route.schema
+
+      if (schema && schema.hide) {
+        continue
+      }
+
+      if (schema && schema.tags && schema.tags.includes(hiddenTag)) {
+        continue
+      }
+
+      const path = stripBasePath
+        ? stripBasePathByServers(route.url, openapiObject.servers)
+        : route.url
+      const url = formatParamUrl(path)
+
+      const swaggerRoute = openapiObject.paths[url] || {}
+
+      const swaggerMethod = {}
+      const parameters = []
+
+      // route.method should be either a String, like 'POST', or an Array of Strings, like ['POST','PUT','PATCH']
+      const methods = typeof route.method === 'string' ? [route.method] : route.method
+
+      for (const method of methods) {
+        swaggerRoute[method.toLowerCase()] = swaggerMethod
+      }
+
+      // All the data the user can give us, is via the schema object
+      if (schema) {
+        // the resulting schema will be in this order
+        if (schema.operationId) {
+          swaggerMethod.operationId = schema.operationId
+        }
+
+        if (schema.tags) {
+          swaggerMethod.tags = schema.tags
+        }
+
+        if (schema.summary) {
+          swaggerMethod.summary = schema.summary
+        }
+
+        if (schema.description) {
+          swaggerMethod.description = schema.description
+        }
+
+        if (schema.externalDocs) {
+          swaggerMethod.externalDocs = schema.externalDocs
+        }
+
+        if (schema.querystring) {
+          getCommonParams('query', parameters, schema.querystring, ref, openapiObject.components.schemas)
+        }
+
+        if (schema.body) {
+          swaggerMethod.requestBody = {
+            content: {}
+          }
+          getBodyParams(swaggerMethod.requestBody.content, schema.body, schema.consumes, ref)
+        }
+
+        if (schema.params) {
+          getCommonParams('path', parameters, schema.params, ref, openapiObject.components.schemas)
+        }
+
+        if (schema.headers) {
+          getCommonParams('header', parameters, schema.headers, ref, openapiObject.components.schemas)
+        }
```

Comment:
Openapi has also option to describe cookies: https://swagger.io/docs/specification/describing-parameters/#cookie-parameters.
Maybe it worth to add them as well. 
```js
        if (schema.cookie) {
          getCommonParams('cookie', parameters, schema.cookie, ref, openapiObject.components.schemas)
        }
```

Reply:
I will treat the input the same as ```querystring``` and we need to document it. As fastify it self do not support ```cookies``` field schema.